src/duty/attestation/attestation.metrics.ts

```diff
@@ -0,0 +1,208 @@
+import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+
+import { ConfigService } from 'common/config';
+import { PrometheusService, TrackTask } from 'common/prometheus';
+import { RegistryService, RegistrySourceOperator } from 'common/validators-registry';
+import { ClickhouseService } from 'storage/clickhouse';
+
+enum BadAttReason {
+  HighIncDelay = 'high_inclusion_delay',
+  InvalidHead = 'invalid_head',
+  InvalidTarget = 'invalid_target',
+  InvalidSource = 'invalid_source',
+}
+
+@Injectable()
+export class AttestationMetrics {
+  protected readonly epochInterval;
+  protected epoch: bigint;
+  protected operators: RegistrySourceOperator[];
+  public constructor(
+    @Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService,
+    protected readonly config: ConfigService,
+    protected readonly prometheus: PrometheusService,
+    protected readonly registryService: RegistryService,
+    protected readonly storage: ClickhouseService,
+  ) {
+    this.epochInterval = this.config.get('BAD_ATTESTATION_EPOCHS');
+  }
+
+  @TrackTask('calc-attestation-metrics')
+  public async calculate(epoch: bigint, possibleHighRewardValidators: string[]) {
+    this.logger.log('Calculating attestation metrics');
+    this.epoch = epoch;
```

Comment:
Slightly discouraged here, may be a more precise field's name should be chosen.

Reply:
fixed